### PR TITLE
Huge PR to bring the model to the current stage of the BMI study.

### PR DESCRIPTION
**Pull request by mfbenitezp**
- New functions in OCL code
- Lot of modifications to run the new yml file, later we will replace that structure for the other areas.
- The new_paramaters.yml file is the one to use it (temporally) 
- The model runs in headless and GUI mode using `python headless.py -p config/new_parameters.yml `

### CODE REVIEW

File: aspics/snapshot.py [hunk 1]
```diff
@@ -57,15 +57,17 @@ def seed_prngs(self, seed):
...
-    def switch_to_healthier_population(self):
-        """
-        Updates to a healthier population by reducing obesity. Any individuals that are overweight or obese are moved
-        to the level of obesity below their current one, by subtracting 1.
-        """
-        people_obesity = self.buffers.people_obesity
-        # only change people with obesity 2 and above, 2 corresponds to "Obese I"
-        people_obesity[people_obesity >= 2] -= 1
-        self.buffers.people_obesity[:] = people_obesity
+    # TODO validate with Hadried that now we dont need this function.
```
[the remark this answers is not in the document]
Comment (HSalat) [reply aspics/snapshot.py:60]: yes

File: aspics/activity.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
 class Activity(enum.Enum):
```
[the remark this answers is not in the document]
Comment (dabreegster) [reply aspics/activity.py:4]: Activity is only used by the OpenCL code in one place, look for "adjust non-home activity flows by the chosen multiplier, while summing the new flows so we can calculate the new home flow". It just checks if the activity is Home or not. Problem is, OpenCL + Python encode home as 0, and SPC as 3. Tracing through, the input buffer is called `place_activities`, and it gets filled out in `convert_snapshot.py`. So I'll do the translation there.

**Comment by dabreegster** [comment]
OK, I've reviewed most of the code and would be happy to merge this now. @mfbenitezp and @HSalat, please check through the commits I pushed. The changes I made are:

- basic cleanup: remove unused code, auto-format things, remove files that're supposed to be in gitignore
- fix a critical bug mapping the Activity enums between SPC and OpenCL
- update the docs to emphasize that all of the yml param files are now broken, except for one
- make the OpenCL code compile on Linux with Intel drivers. Please check it also works on your Mac. I'm not entirely sure why different driver versions are OK with `int(1)` vs `(int) 1` vs `convert_int(1)`.

I did not review the notebook or `calibration/run_aspics.py` at all. Is that important?

As a next step, I can cleanup the mess of parameter plumbing. The flow is .yml -> `loader.py` -> `params.py` -> OpencL, and it's a complete mess right now. Once we decide where default values for params common between study areas should live, I can make this much simpler.

**Comment by dabreegster** [comment]
Will remove old broken yaml files from this PR and just keep ones that work with the current code. The others can be restored later, and of course they're still in git

**Comment by mfbenitezp** [comment]
Can we rebase and merge this? @dabreegster I'm ok with all the updates made by u.

**Comment by dabreegster** [comment]
Yup, I'll clean up the old YAML files and merge after lunch

**Pull request merged**